doc/reference/trd-legal.md

```diff
@@ -0,0 +1,188 @@
+
+This document describes Tock’s policy on licensing and copyright. It explains
+the rationale behind the license selection (dual-license, MIT or Apache2) and
+copyright policy (optional, and authors may retain copyright if desired). It
+further outlines how licensing and copyright shall be handled throughout
+projects under the Tock umbrella.
+
+Explicitly not discussed in this TRD are issues of trademark, the Tock brand,
+logos, or other non-code assets and artifacts.
+
+
+1. Introduction
+===============
+
+Tock’s goal is to provide a safe, secure, and efficient operating system for
+microcontroller-class devices. The Tock project further believes it is
+important to enable and support widespread adoption of safe, secure, and
+efficient software. Tock also seeks to be an open and inclusive project, and
+Tock welcomes contributions from any individuals or entities wishing to
+improve the safety, security, reliability, efficiency, or usability of Tock
+and the Tock ecosystem.
+
+The intent of these policies is to best satisfy the needs of all stakeholders
+in the Tock ecosystem.
+
+
+2. Licensing
+============
+
+Tock’s primary licensing goal is to ensure ease of use for Tock project elements.
+For this reason, Tock offers its resources under the minimally encumbered MIT
+license and under the Apache2 license for those entities concerned about possible
+litigation issues.
+
+All software artifacts under the umbrella of the Tock project are
+dual-licensed as Apache License, Version 2.0 (LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0) or MIT license (LICENSE-MIT or
+http://opensource.org/licenses/MIT).
+
+All contributions to Tock projects MUST be licensed under these terms.
+
+
+3. Copyright
+============
+
+Entities contributing resources to open-source projects often require attribution
+to recognize their efforts. Copyright is a common means to provide this. For
+downstream users, the license terms of Tock ensures unencumbered use.
+
+Copyrights in Tock projects are retained by their contributors. No
+copyright assignment is required to contribute to Tock projects.
+
+Artifacts in the Tock project MAY include explicit copyright notices.
+Substantial updates to an artifact MAY add additional copyright notices
+to an artifact. Copyright notices MUST NOT be removed by any party except
+the original copyright holder, and removal is done solely at their discretion.
+
+For full authorship information, see the version control history.
+
+
+4. Implementation
+=================
+
+All code artifacts in Tock projects MUST include a license notice.
+
+Code artifacts MAY include copyright notice(s). Copyright notices MUST
+include a year. Newer copyright notices MUST be placed after existing
+copyright notices. If non-trivial updates are performed by an original
+copyright author, they MAY amend the year(s) indicated on their existing
+copyright statement or MAY add an additional copyright line, at their
+discretion.
+
+
+4.1 Format
```

Comment:
The entire "Format" section makes it difficult to re-use code from other projects, as their license and copyright information may not match our requirements, and I do not think we can arbitrarily reformat those notices.

An example of a file that does not match this format is [`capsules/src/sip_hash.rs`](https://github.com/tock/tock/blob/0f7fe7d9355002e4e7065cc3fb940a0d121e8d21/capsules/src/sip_hash.rs).